dev-util/bpftool/bpftool-7.5.0-r1.ebuild

```diff
@@ -65,8 +67,7 @@ src_unpack() {
 	if [[ -n ${LINUX_PATCH} ]] ; then
 		eshopts_push -o noglob
 		ebegin "Filtering partial source patch"
-		filterdiff -p1 ${paths[@]/#/-i } -z "${DISTDIR}"/${LINUX_PATCH} \
-			> ${P}.patch
+		xzcat "${DISTDIR}"/${LINUX_PATCH} | filterdiff -p1 ${paths[@]/#/-i} > ${P}.patch
 		eend $? || die "filterdiff failed"
```

Comment:
This needs to become `assert` instead to catch pipe failure.

Reply:
Didn't know about assert, thanks. Added a "test -s" to verify that there is some non-zero output in ${P}. Tested by using cat instead of xzcat, which explodes as expected.
Resolve if that was what you had in mind.

Reply:
LGTM, but let's do `nonfatal assert -n` so that the `eend $?` afterward works.

Reply:
> LGTM, but let's do `nonfatal assert -n` so that the `eend $?` afterward works.

Sorry for catching this only now. `assert -n` will die in spite of the `-n` option because it is not called under `nonfatal`.

The suggested `nonfatal assert -n` would not work either because the `nonfatal` command resets the `PIPESTATUS` variable (so `assert` would always return good status).

The way forward is to replace it with the `pipestatus` command that will be available soon.

Reply:
Just FYI that this bpftool ebuild has been removed already (for other reasons), but your comment still applies to the current perf-6.12 ebuild.

Reply:
> Just FYI that this bpftool ebuild has been removed already (for other reasons), but your comment still applies to the current perf-6.12 ebuild.

Yeah, and no need to followup for this ebuild separately. Later I'll submit a global PR fixing all occurences.